torchtitan/utils.py

```diff
@@ -145,7 +148,12 @@ def get_peak_flops(device_name: str) -> int:
             return 1979e12
         elif "PCIe" in device_name:
             return 756e12
-        else:  # for SXM and other variants
+        # Sometimes, the device name is just "NVIDIA H100", we need
+        # to check the memory size to determine the peak flops.
+        # H100 NVL
```

Comment:
ideally we should still include pointers to how these numbers were obtained... if possible

Reply:
L147 is never hit because like I said in the comment, device name is literally "NVIDIA H100" there is no "NVL" or "SXM" in its name. So L147 is never hit in our case.

Reply:
The number comes from the same source you listed in L145. The only single source is L145.